promql/rawchunktest/rawchunk_test.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rawchunktest
```

Comment:
You could also leave this file up in the promql directory and put it into the `promql_test` package. The extra directory just for one test is weird.

Reply:
ack, will do, also I'll add something in storage as you suggest

Reply:
moved to engine_test.go

Reply:
added test in buffer_test.go